cmd/editorconfig-checker/main_test.go

```diff
@@ -133,6 +135,52 @@ func TestMainDryRun(t *testing.T) {
 	}
 }
 
+func TestMainColorSupport(t *testing.T) {
+	type env map[string]string
+	type args []string
+
+	tests := []struct {
+		name string
+		env  env
+		args args
+	}{
+		{"no-envvar-no-arg", env{}, args{}},
+		{"envvar-no-arg", env{"NO_COLOR": "1"}, args{}},
+		{"no-envvar-color-off", env{}, args{"--no-color"}},
+		{"no-envvar-color-on", env{}, args{"--color"}},
+		// test that arguments win over env var
+		{"envvar-color-off", env{"NO_COLOR": "1"}, args{"--no-color"}},
+		{"envvar-color-on", env{"NO_COLOR": "1"}, args{"--color"}},
+		// test that the last argument wins
+		{"no-envvar-color-offon", env{}, args{"--no-color", "--color"}},
+		{"no-envvar-color-onoffon", env{}, args{"--color", "--no-color", "--color"}},
+		// test that boolean values in the env var work correctly
+		{"envvar-true", env{"NO_COLOR": "true"}, args{}},
+		{"envvar-false", env{"NO_COLOR": "false"}, args{}},
+		{"envvar-zero", env{"NO_COLOR": "0"}, args{}},
+		// test that pure string values are correctly handled as setting NoColor
+		{"envvar-yes", env{"NO_COLOR": "yes"}, args{}},
+		{"envvar-no", env{"NO_COLOR": "no"}, args{}},
```

Comment:
```suggestion
		// test that unsupported string values are handled as NoColor=1
		{"envvar-yes", env{"NO_COLOR": "yes"}, args{}},
		{"envvar-no", env{"NO_COLOR": "no"}, args{}},
		{"envvar-foo", env{"NO_COLOR": "foo"}, args{}},
```

I also added a test, that was missing I think

Reply:
I added a bit more ridiculous version of that. But essentially that last test case also just exercises the path that the values `yes` and ̀`no` would take.